compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -2343,6 +2343,21 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
       def unapply(x: NoPrefix): true = true
     end NoPrefix
 
+    type FlexibleType = dotc.core.Types.FlexibleType
+
+    object FlexibleTypeTest extends TypeTest[TypeRepr, FlexibleType]:
+      def unapply(x: TypeRepr): Option[FlexibleType & x.type] = x match
+        case x: (Types.FlexibleType & x.type) => Some(x)
+        case _ => None
+    end FlexibleTypeTest
+
+    object FlexibleType extends FlexibleTypeModule:
+      def apply(tp: TypeRepr): FlexibleType =
+        // why object FlexibleType.apply(Type) is not found here?
+        Types.FlexibleType(tp, tp)
+      def unapply(x: FlexibleType): TypeRepr = x.hi
```

Comment:
The type cannot be tested at call site before calling the unapply. The type test must be done in the unapply implementation. See other examples above.

```suggestion
      def unapply(x: TypeRepr): Option[TypeRepr & x.type] = 
         ...
```